common/web/lm-worker/src/correction/distance-modeler.ts

```diff
@@ -537,6 +535,70 @@ namespace correction {
         maxTime = waitMillis;
       }
 
+      class ExecutionTimer {
+        private start: number;
+        private loopStart: number;
+        private maxExecutionTime: number;
+        private maxTrueTime: number;
+
+        private executionTime: number;
+
+        /**
+         * Used to track intervals in which potential context swaps by the OS may
+         * have occurred.
+         */
```

Comment:
I'd like to see some more detail in this comment for future maintainers -- how this works, how it tracks the 'context swap' time (how do we know that's what it is tracking?) etc.